src/aatoolbox/datasources/chirps/chirps.py

```diff
@@ -0,0 +1,572 @@
+        ).tolist()
+
+        # Create a sentence containing information on the downloaded data
+        sentence_to_print = (
+            f"{self._frequency.capitalize()} "
+            "data will be downloaded, starting from "
+            f"{date_list[0]} to {date_list[-1]}."
+        )
+
+        return sorted(set(date_list)), sentence_to_print
+
+    def _check_dates_validity(self, start_date, end_date):
+        """Check dates vailidity."""
+        start_avail_date = date(year=1981, month=1, day=1)
+        end_avail_date = self._get_last_available_date()
+
+        if not start_avail_date <= start_date <= end_date <= end_avail_date:
+            raise ValueError(
+                "Make sure that the input dates are ordered in the "
+                f"following way: {start_avail_date} <= {start_date} "
+                f"<= {end_date} <= {end_avail_date}. The two dates above "
+                "indicate the range for which CHIRPS data are "
+                "currently available."
+            )
+
+    def _get_file_name_base(
+        self,
+        year: Optional[str],
+        month: Optional[str],
+        day: Optional[str],
+    ) -> str:
+        # Set wildcard for year, to get general filename pattern
+        # used to find all files with pattern in folder
+        if year is None:
+            year = "*"
+        if month is None:
+            month = "*"
+        if day is None:
+            day = "*"
+
+        file_name_base = (
+            f"{self._country_config.iso3}_chirps_"
+            f"{self._frequency}_{year}_{month}_"
+        )
+
+        return file_name_base
+
+    def _get_file_name(
+        self,
+        year: Optional[str],
+        month: Optional[str],
+        day: Optional[str],
+    ) -> str:
+        pass
+
+    def _get_raw_path(
+        self, year: Optional[str], month: Optional[str], day: Optional[str]
+    ) -> Path:
+        return self._raw_base_dir / self._get_file_name(
+            year=year, month=month, day=day
+        )
+
+    def _get_processed_path(self, raw_path: Path) -> Path:
+        return self._processed_base_dir / raw_path.parts[-1]
+
+    def _get_base_url(self):
+
+        base_url = (
+            "https://iridl.ldeo.columbia.edu/SOURCES/.UCSB/.CHIRPS/.v2p0/"
+        )
+
+        location_url = (
+            f"X/%28{self._geobb.lon_min}%29%28{self._geobb.lon_max}"
+            f"%29RANGEEDGES/"
+            f"Y/%28{self._geobb.lat_max}%29%28{self._geobb.lat_min}"
+            f"%29RANGEEDGES/"
+        )
+
+        return base_url, location_url
+
+    def _get_url(self, year: str, month: str, day: str):
+        pass
+
+    def _get_last_available_date(self):
+        pass
+
+    def _get_downloaded_path_list(self):
+        """Create a list with all raw data downloaded."""
+        # Get the path where raw data is stored
+        raw_path = self._get_raw_path(year=None, month=None, day=None)
+        return list(raw_path.parents[0].glob(raw_path.name))
```

Comment:
I think it'd be handy to keep it as it is for the method `process()`. This step requires very little time to run, so, instead of having the user needing to specify `start_date` end `end_date` for the method, it may be just easier to process all downloaded data. What do you think?

In the `load()` method I agree with you, I indeed changed it.

Reply:
I was trying to think of what we do in other cases, and I realized that for NDVI the start and end dates are [passed to the constructor](https://github.com/OCHA-DAP/pa-aa-toolbox/pull/100/files#diff-f240ac32a46252d1b17ad5c1779b358f314b935636d9629660270b30b86384e7R77). (So sorry, I should have checked that earlier). What do you think of doing that instead?